software/deploy/deploy.go

```diff
@@ -202,15 +204,30 @@ func UpdateDeploymentImage(houstonClient houston.ClientInterface, deploymentID,
 	return deploymentID, err
 }
 
-func pushDockerImage(byoRegistryEnabled bool, byoRegistryDomain, name, nextTag, cloudDomain string, imageHandler airflow.ImageHandler, houstonClient houston.ClientInterface, c *config.Context, customImageName string) error {
+func pushDockerImage(byoRegistryEnabled bool, deploymentInfo *houston.Deployment, byoRegistryDomain, name, nextTag, cloudDomain string, imageHandler airflow.ImageHandler, houstonClient houston.ClientInterface, c *config.Context, customImageName string) error {
 	var registry, remoteImage, token string
 	if byoRegistryEnabled {
 		registry = byoRegistryDomain
 		remoteImage = fmt.Sprintf("%s:%s", registry, fmt.Sprintf("%s-%s", name, nextTag))
 	} else {
-		registry = registryDomainPrefix + cloudDomain
-		remoteImage = fmt.Sprintf("%s/%s", registry, airflow.ImageName(name, nextTag))
-		token = c.Token
+		platformVersion, _ := houstonClient.GetPlatformVersion(nil)
+		if versions.GreaterThanOrEqualTo(platformVersion, "1.0.0") {
+			registry, err := getDeploymentRegistryUrl(deploymentInfo.Urls)
+			if err != nil {
+				return err
+			}
+			// Switch to per deployment registry login
+			err = auth.RegistryAuth(houstonClient, os.Stdout)
+			if err != nil {
+				logger.Debugf("There was an error logging into registry: %s", err.Error())
```

Comment:
Why do we not exit here? It's not going to work if we haven't auth'd right?

Reply:
in existing code as well even if the login fails on push it will get into error - thats why we didnt change this, now should we make this to return error with fail even when the auth gets failed

Reply:
Returning the `err` here is the right call. Better to fail earlier if we know it isn't going to work.